### ISSUE TITLE
Support 'podman'

### ISSUE THREAD

**Issue by mih** [datalad/datalad-container#89]
https://podman.io

> What is Podman? Podman is a daemonless container engine for developing, managing, and running OCI Containers on your Linux System. Containers can either be run as root or in rootless mode. Simply put: `alias docker=podman`.

**Comment by mih** [comment]
DEB packages are available from http://ppa.launchpad.net/projectatomic

**Comment by almereyda** [comment]
I would believe the work requested here only requires little adaptation for rootless containers, and could also benefit to Docker rootless.

If volumes are mapped, for example, it will be good to use the `:z` or `:Z` mount options.

Otherwise this should also work as a drop-in replacement and also removes the limitation currently present in the foot note of:

* [7.2. Computational reproducibility with software containers — The DataLad Handbook](https://handbook.datalad.org/en/latest/basics/101-133-containersrun.html#id1)

**Comment by yarikoptic** [comment]
indeed, podman is gaining popularity and even for HPC (see e.g. https://github.com/NERSC/podman-hpc although I did not look into what it does there exactly). I guess similarly to docker ideally we should look also into addressing #98 first, likely through using another mechanism than we have ATM for how we place layers on a filesystem.
